pypit/spectrographs/wavelengths.py

```diff
@@ -0,0 +1,208 @@
+""" Spectrograph specific info/methods for wavelengths
+"""
+import numpy as np
+
+from astropy.io import fits
+
+from pypit import msgs
+from pypit import arparse
+from pypit import ararclines
+
+from pypit import ardebug as debugger
+
+
+def setup_param(spectrograph, msarc_shape, fitstbl, arc_idx,
+                calibrate_lamps=None):
+    """ Setup for arc analysis
+
+    Parameters
+    ----------
+    spectrograph : str
+    msarc_shape : tuple
+    fitstbl : Table
+      Contains relevant information from fits header files
+    arc_idx : int
+      Index of the arc frame in the fitstbl
+    calibrate_lamps : str, optional
+       List of lamps used
+
+    """
+    # Defaults
+    arcparam = dict(llist='',
+                    disp=0.,             # Ang/unbinned pixel
+                    b1=0.,               # Pixel fit term (binning independent)
+                    b2=0.,               # Pixel fit term
+                    lamps=[],            # Line lamps on
+                    wv_cen=0.,           # Estimate of central wavelength
+                    wvmnx=[2900.,12000.],# Guess at wavelength range
+                    disp_toler=0.1,      # 10% tolerance
+                    match_toler=3.,      # Matching tolerance (pixels)
+                    min_ampl=300.,       # Minimum amplitude
+                    func='legendre',     # Function for fitting
+                    n_first=1,           # Order of polynomial for first fit
+                    n_final=4,           # Order of polynomial for final fit
+                    nsig_rej=2.,         # Number of sigma for rejection
+                    nsig_rej_final=3.0,  # Number of sigma for rejection (final fit)
+                    Nstrong=13)          # Number of lines for auto-analysis
+
+    modify_dict = None
+    # Instrument/disperser specific
+    disperser = fitstbl["dispname"][arc_idx]
+    binspatial, binspectral = arparse.parse_binning(fitstbl['binning'][arc_idx])
+    if spectrograph == 'shane_kast_blue':
```

Comment:
You know what I'm going to say... these should all go in the relevant `spectroclass` for each instrument  :-)

... and then, we wouldn't need the `wavelengths.py` file

Reply:
on it.